src/group_theory/subgroup.lean

```diff
@@ -1186,6 +1186,23 @@ begin
   simp only [],
 end
 
+@[to_additive] lemma map_eq_bot_iff (f : G →* N) {H : subgroup G} : H.map f = ⊥ ↔ H ≤ f.ker :=
+begin
+  rw eq_bot_iff,
+  split,
+  { exact λ h x hx, h ⟨x, hx, rfl⟩ },
+  { intros h x hx,
+    obtain ⟨y, hy, rfl⟩ := hx,
+    exact h hy },
+end
+
+@[to_additive] lemma ker_eq_bot_iff (f : G →* N) : f.ker = ⊥ ↔ function.injective f :=
+begin
+  split,
+  { intros h x y hxy,
+    rwa [←mul_inv_eq_one, ←map_inv, ←map_mul, ←mem_ker, h, mem_bot, mul_inv_eq_one] at hxy },
+  { exact λ h, le_bot_iff.mp (λ x hx, h (hx.trans f.map_one.symm)) },
+end
```

Comment:
I think the first one of these should be in the subgroup namespace (with an explicit `H` argument, so it can be used as `H.map_eq_bot_iff`), while the second is in the right place already.

Reply:
Done.